core/domain/fs_domain.py

```diff
@@ -108,96 +108,98 @@ def version(self):
 
-    The conceptual intention is for each exploration to have its own asset
-    folder. An asset has no meaning outside its exploration, so the assets in
+    The conceptual intention is for each entity to have its own asset
+    folder. An asset has no meaning outside its entity, so the assets in
     these asset folders should therefore not be edited directly. They should
     only be modified as side-effects of some other operation (such as adding an
-    image to an exploration).
+    image to an entity like topics, stories or explorations).
 
-    The content of an exploration should include a reference to the asset
+    The content of an entity should include a reference to the asset
     together with the version number of the asset. This allows the
-    exploration to refer to asset versions.
+    entity to refer to asset versions.
 
     In general, assets should be retrieved only within the context of the
-    exploration that contains them, and should not be retrieved outside this
+    entity that contains them, and should not be retrieved outside this
     context.
 
     Args:
-        exploration_id: str. The id of the exploration.
+        assets_path: str. The path to the assets folder of the corresponding
+            entity. It is of the form {{entity}}/{{entity_id}}
+            (eg: exploration/{{exp_id}})
     """
 
     _DEFAULT_VERSION_NUMBER = 1
 
-    def __init__(self, exploration_id):
-        """Constructs a ExplorationFileSystem object.
+    def __init__(self, assets_path):
+        """Constructs a EntityFileSystem object.
 
         Args:
-            exploration_id: str. The id of the exploration.
+            assets_path: str. The path to the parent directory for all assets.
         """
-        self._exploration_id = exploration_id
+        self._assets_path = assets_path
 
     @property
-    def exploration_id(self):
-        """Returns the id of the exploration.
+    def assets_path(self):
+        """Returns the path of the parent folder of assets.
 
         Returns:
-            str. The exploration id.
+            str. The path.
         """
-        return self._exploration_id
+        return self._assets_path
 
     def _get_file_metadata(self, filepath, version):
         """Return the desired file metadata.
 
         Returns None if the file does not exist.
 
         Args:
-            filepath: str. The path to the relevant file within the exploration.
+            filepath: str. The path to the relevant file within the entity.
             version: int. The version number of the file whose metadata is to be
                 returned.
 
         Returns:
             FileMetadataModel or None. The model instance representing the file
-                metadata with the given exploration_id, filepath, and version,
+                metadata with the given assets_path, filepath, and version,
                 or None if the file does not exist.
         """
         if version is None:
```

Comment:
Do we even need the "version" stuff?

Reply:
I think so, as this function is called by the get function, and it has a versions parameter.